All_home_works/hw8/task2.py

```diff
@@ -0,0 +1,64 @@
+"""
+Description in task2.rst
+"""
+import sqlite3
+
+
+class Data:
+    def __init__(self, cursor):
+        self.cursor = cursor
+        self.column_names = [x[0] for x in self.cursor.description]
+
+    def __next__(self):
+        try:
+            if data := self.cursor.fetchone():
+                self.set_attributes_from_row(self.column_names, data)
+                return self
+            raise StopIteration
+        except sqlite3.ProgrammingError:
+            raise StopIteration
+
+    def set_attributes_from_row(self, column_names, row):
+        for key, value in zip(column_names, row):
+            setattr(self, key, value)
+
+    def __getitem__(self, item):
+        return self.__dict__[item]
+
+    def __eq__(self, other):
+        return other in self.__dict__.values()
+
+
+class TableData:
+    def __init__(self, db_name, db_table):
+        self.conn = sqlite3.connect(db_name)
+        self.cursor = self.conn.cursor()
+        self.db_table = db_table
+
+    def __iter__(self):
+        self.cursor.execute(f"SELECT * FROM '{self.db_table}'")  # noqa: S608
+        return Data(cursor=self.cursor)  # noqa: S608
+
+    def __len__(self):
+        self.cursor.execute(
+            f"SELECT count(*) FROM '{self.db_table}' query"  # noqa: S608
+        )
+        return self.cursor.fetchone()[0]
+
+    def __getitem__(self, item):
+        self.cursor.execute(
+            f"SELECT * FROM '{self.db_table}' where name='{item}'"  # noqa: S608
```

Comment:
What could go wrong here?

Reply:
- Maybe there is no such table
- Maybe there is no such row in the table
- Maybe there are some rows with such name, not one

Reply:
`Select *` It would give us all fields at a table. (password, auth_token, e.c.)

Reply:
What else could be bad here?  hint: `You know that thing`